### PR TITLE
optimizate controllers with one sharedInformerFactory

### LINKED ISSUE #3484: [Performance issue] All controllers should use sharedInformerFactory
**What happened**:
Currently, every controller in volcano (like job controller, queue controlelr, jobflow, etc.) uses a seperate vcinformerFactory, which causes memory wastage, we should reduce memory usage using sharedInformerFactory.
**What you expected to happen**:
Use sharedInformerFactory in controller and just initiate one sharedInformerFactory like what k8s sharedInformerFactory done.
**How to reproduce it (as minimally and precisely as possible)**:
None

**Anything else we need to know?**:
1、where controllers use a newInformerFactory everytime.
https://github.com/volcano-sh/volcano/blob/aa7c8b8b390cc60b1fd5961be81773d419cc5bab/pkg/controllers/job/job_controller.go#L155 
2、we should also add a vcsharedInformerFactory here.
https://github.com/volcano-sh/volcano/blob/aa7c8b8b390cc60b1fd5961be81773d419cc5bab/pkg/controllers/framework/interface.go#L30
**Environment**:
- Volcano Version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by y-ykcir**
fix #3484

### CODE REVIEW

File: pkg/controllers/jobtemplate/jobtemplate_controller.go [hunk 1]
```diff
@@ -81,14 +80,14 @@ func (jt *jobtemplatecontroller) Initialize(opt *framework.ControllerOption) err
 	jt.kubeClient = opt.KubeClient
 	jt.vcClient = opt.VolcanoClient
 
-	jt.jobTemplateInformer = informerfactory.NewSharedInformerFactory(jt.vcClient, 0).Flow().V1alpha1().JobTemplates()
+	jt.jobTemplateInformer = opt.VCSharedInformerFactory.Flow().V1alpha1().JobTemplates()
```
Comment (Monokaix) [inline pkg/controllers/jobtemplate/jobtemplate_controller.go:83]: Welcome! Please change to `factory := opt.VCSharedInformerFactory` to keep consistent with other controllers.

File: pkg/controllers/jobflow/jobflow_controller.go [hunk 2]
```diff
@@ -89,19 +88,19 @@ func (jf *jobflowcontroller) Initialize(opt *framework.ControllerOption) error {
 	jf.kubeClient = opt.KubeClient
 	jf.vcClient = opt.VolcanoClient
 
-	jf.jobFlowInformer = informerfactory.NewSharedInformerFactory(jf.vcClient, 0).Flow().V1alpha1().JobFlows()
+	jf.jobFlowInformer = opt.VCSharedInformerFactory.Flow().V1alpha1().JobFlows()
```
Comment (Monokaix) [inline pkg/controllers/jobflow/jobflow_controller.go:91]: So does here.

File: pkg/controllers/jobtemplate/jobtemplate_controller.go (hunk 1 above)
Comment (y-ykcir) [reply pkg/controllers/jobtemplate/jobtemplate_controller.go]: done

File: pkg/controllers/jobflow/jobflow_controller.go (hunk 2 above)
Comment (y-ykcir) [reply pkg/controllers/jobflow/jobflow_controller.go]: done

**Comment by william-wang** [comment]
@y-ykcir The `code verify` failed, please make it happy:)

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**